encord/constants/model.py

```diff
@@ -13,14 +13,76 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-FAST_AI = "fast_ai"
-RESNET18 = "resnet18"
-RESNET34 = "resnet34"
-RESNET50 = "resnet50"
-RESNET101 = "resnet101"
-RESNET152 = "resnet152"
-VGG16 = "vgg16"
-VGG19 = "vgg19"
-YOLOV5 = "yolov5"
-FASTER_RCNN = "faster_rcnn"
-MASK_RCNN = "mask_rcnn"
+from __future__ import annotations
+
+from enum import Enum
+from typing import List
+
+
+class AUTOMATION_MODELS(Enum):
+    FAST_AI = "fast_ai"
+    RESNET18 = "resnet18"
+    RESNET34 = "resnet34"
+    RESNET50 = "resnet50"
+    RESNET101 = "resnet101"
+    RESNET152 = "resnet152"
+    VGG16 = "vgg16"
+    VGG19 = "vgg19"
+    YOLOV5 = "yolov5"
+    FASTER_RCNN = "faster_rcnn"
+    MASK_RCNN = "mask_rcnn"
+
+    @staticmethod
+    def classification_options() -> List[AUTOMATION_MODELS]:
+        """
+        Returns:
+            Model types that can be used for frame-level classifications.
+            That is, model types that can be used with ``<feature_node_hash>``es from
+            the ``classification`` part of the ontology.
+        """
+        return [
+            AUTOMATION_MODELS.FAST_AI,
+            AUTOMATION_MODELS.RESNET18,
+            AUTOMATION_MODELS.RESNET34,
+            AUTOMATION_MODELS.RESNET50,
+            AUTOMATION_MODELS.RESNET101,
+            AUTOMATION_MODELS.RESNET152,
+            AUTOMATION_MODELS.VGG16,
+            AUTOMATION_MODELS.VGG19,
+        ]
+
+    @staticmethod
+    def object_detection_options() -> List[AUTOMATION_MODELS]:
+        """
+        Returns:
+            Model types that can be used with bounding_box type
+            ``<feature_node_hashes>`` from the ``objects`` part of the project ontology.
+        """
+        return [AUTOMATION_MODELS.YOLOV5, AUTOMATION_MODELS.FASTER_RCNN]
+
+    @staticmethod
+    def instance_segmentation_options() -> List[AUTOMATION_MODELS]:
+        """
+        Returns:
+            Model types that can be used with polygon type ``<feature_node_hashes>``
+            from the ``objects`` part of the project ontology.
+        """
+        return [AUTOMATION_MODELS.MASK_RCNN]
+
+    @staticmethod
+    def has_value(cls, value):
+        return value in set(v for v in cls._value2member_map_)
```

Comment:
would `return value in cls._value2member_map_` work?

Feels a bit scary to expose the private member here as I think there is no API guarantee, but probably fine.

Do you have access to our SDK integration tests? (in the cord-backend). Would probably be worth changing a test there too.